src/user/models/kyc/kyc-webhook.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { UserDataRepository } from '../user-data/user-data.repository';
+import { HttpService } from 'src/shared/services/http.service';
+import { WalletRepository } from '../wallet/wallet.repository';
+import { KycCompleted } from '../user-data/user-data.entity';
+import { Config } from 'src/config/config';
+
+@Injectable()
+export class KycWebhookService {
+  constructor(
+    private readonly userDataRepo: UserDataRepository,
+    private readonly http: HttpService,
+    private readonly walletRepo: WalletRepository,
+  ) {}
+
+  async kycChanged(userDataId: number): Promise<void> {
+    let data: { id?: string; result?: 'StatusChanged' | 'Failed'; data?: {}; reason?: string } = {};
```

Comment:
It would make sense to add interfaces for this:

```ts
export enum KycStatus {
  NA = 'NA',
  LIGHT = 'Light',
  FULL = 'Full',
}

export enum KycResult {
  STATUS_CHANGED = 'StatusChanged',
  FAILED = 'Failed',
}

export class KycDataDto {
  mail: string;
  firstName: string;
  lastName: string;
  street: string;
  houseNumber: string;
  city: string;
  zip: string;
  phone: string;
  kycStatus: KycStatus;
  kycHash: string;
}

export class KycWebhookDto {
  id: string;
  result: KycResult;
  data: KycDataDto;
  reason: string;
}
```

Reply:
Top, I would named it KycWebhookStatus but yeah makes sense.